src/main/java/org/gelecekbilimde/scienceplatform/auth/service/impl/RoleSelfApplicationServiceImpl.java

```diff
@@ -0,0 +1,133 @@
+package org.gelecekbilimde.scienceplatform.auth.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.gelecekbilimde.scienceplatform.auth.exception.RoleApplicationAlreadyExistException;
+import org.gelecekbilimde.scienceplatform.auth.exception.RoleApplicationNotFoundByUserIdAndStatusException;
+import org.gelecekbilimde.scienceplatform.auth.exception.RoleNotFoundByNameException;
+import org.gelecekbilimde.scienceplatform.auth.model.Identity;
+import org.gelecekbilimde.scienceplatform.auth.model.RoleApplication;
+import org.gelecekbilimde.scienceplatform.auth.model.RoleSelfApplicationFilter;
+import org.gelecekbilimde.scienceplatform.auth.model.entity.RoleApplicationEntity;
+import org.gelecekbilimde.scienceplatform.auth.model.entity.RoleEntity;
+import org.gelecekbilimde.scienceplatform.auth.model.enums.RoleApplicationStatus;
+import org.gelecekbilimde.scienceplatform.auth.model.enums.RoleName;
+import org.gelecekbilimde.scienceplatform.auth.model.mapper.RoleApplicationEntityToDomainMapper;
+import org.gelecekbilimde.scienceplatform.auth.model.request.RoleSelfApplicationListRequest;
+import org.gelecekbilimde.scienceplatform.auth.repository.RoleApplicationRepository;
+import org.gelecekbilimde.scienceplatform.auth.repository.RoleRepository;
+import org.gelecekbilimde.scienceplatform.auth.service.RoleSelfApplicationService;
+import org.gelecekbilimde.scienceplatform.common.model.BasePage;
+import org.gelecekbilimde.scienceplatform.user.model.entity.UserEntity;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+class RoleSelfApplicationServiceImpl implements RoleSelfApplicationService {
+
+	private final RoleApplicationRepository roleApplicationRepository;
+	private final RoleRepository roleRepository;
+	private final Identity identity;
+
+
+	private final RoleApplicationEntityToDomainMapper roleApplicationEntityToDomainMapper = RoleApplicationEntityToDomainMapper.initialize();
```

Comment:
@coderabbitai The area here cannot be injected by spring. Because it's not a bean. In addition, it cannot inject because we use lombok anotations. Don't make this suggestion for mappers.